app/javascript/packages/document-capture/components/form-steps.jsx

```diff
@@ -146,38 +102,45 @@ function FormSteps({ steps = [], onComplete = () => {}, initialValues = {}, init
     didSubmitWithErrors.current = false;
   }, [activeErrors]);
 
-  // An "effective" step is computed in consideration of the facts that (1) there may be no history
-  // parameter present, in which case the first step should be used, and (2) the values may not be
-  // valid for previous steps, in which case the furthest valid step should be set.
-  const effectiveStepIndex = Math.max(
-    Math.min(getStepIndexByName(steps, stepName), getLastValidStepIndex(steps, values) + 1),
-    0,
-  );
-  const effectiveStep = steps[effectiveStepIndex];
-  useEffect(() => {
-    // The effective step is used in the initial render, but since it may be out of sync with the
-    // history parameter, it is synced after mount.
-    if (effectiveStep && stepName && effectiveStep.name !== stepName) {
-      setStepName(effectiveStep.name);
-    }
+  const stepIndex = Math.max(getStepIndexByName(steps, stepName), 0);
+  const step = steps[stepIndex];
 
+  useEffect(() => {
     // Treat explicit initial step the same as step transition, placing focus to header.
-    if (initialStep && headingRef.current) {
+    if (autoFocus && headingRef.current) {
       headingRef.current.focus();
     }
   }, []);
 
+  /**
+   * Returns array of form errors for the current set of values.
+   *
+   * @return {FormStepError<Record<string,Error>>[]}
+   */
+  function getValidationErrors() {
+    return Object.keys(fields.current).reduce((result, key) => {
+      const { element, isRequired } = fields.current[key];
+      const isActive = !!element;
+
+      if (isActive && isRequired && !values[key]) {
+        result = result.concat({ field: key, error: new RequiredValueMissingError() });
+      }
+
+      return result;
+    }, /** @type {FormStepError<Record<string,Error>>[]} */ ([]));
+  }
+
   useEffect(() => {
     // Errors are assigned at the first attempt to submit. Once errors are assigned, track value
     // changes to remove validation errors as they become resolved.
     if (activeErrors) {
-      const nextActiveErrors = getValidationErrors(effectiveStep, values);
+      const nextActiveErrors = getValidationErrors();
```

Comment:
The issue described at https://github.com/18F/identity-idp/pull/4243#issuecomment-700778420 is related to this, where in `toNextStep` we will always assign `setActiveErrors(getValidationErrors())`. In the mobile flow this happens when continuing from the first "Mobile Intro" step. There are no actual errors, but it still sets an empty array, which is considered truthy by `if (activeErrors) {`. The idea with this code was to try to "chip away at" any active errors.

Possible fixes:

- Change `if (activeErrors) {` to `if (activeErrors?.length)`
- Guard `setActiveErrors` in `toNextStep` to only set active errors if non-empty
- Return `undefined` from `getValidationErrors` if result would be empty